### PR TITLE
Eslint Prep Work

### PR DESCRIPTION
**Pull request by webstech**
tslint is being deprecated.  There is some tooling to support the conversion to eslint and some typescript specific linting.  This is an initial pass to update the code for the low hanging fruit.  Best  reviewed at the commit level.  There are very few logic changes.

[typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) has more info about converting.  Initial playing with the conversion required turning indent rules off - the errors and warnings were not consistent.  The typescript-eslint team acknowledged in an issue they do not use the indent rules and use prettier instead.

### CODE REVIEW

File: lib/mail-archive-helper.ts [hunk 1]
```diff
@@ -62,9 +62,8 @@ export class MailArchiveGitHelper {
             return "";
         }
 
-        return "``````````\n" +
-            body + (body.endsWith("\n") ? "" : "\n") +
-            "``````````\n";
+        return "``````````\n"
+            + body + (body.endsWith("\n") ? "" : "\n") + "``````````\n";
```
Comment (dscho) [inline lib/mail-archive-helper.ts]: I guess this cannot be converted to a template? ;-)

File: lib/git.ts [hunk 2]
```diff
@@ -137,7 +138,7 @@ export async function revParse(argument: string, workDir?: string) {
  * @returns number the number of commits in the commit range
  */
 export async function revListCount(rangeArgs: string | string[],
-                                   workDir: string = "."):
+                                   workDir = "."):
```
Comment (dscho) [inline lib/git.ts]: This change probably wanted to go into "remove inferred types"...

File: lib/send-mail.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export interface IParsedMBox {
     cc?: string[];
     date?: string;
     from?: string;
-    headers?: Array<{ key: string; value: string; }>;
+    headers?: Array<{ key: string; value: string }>;
```
Comment (dscho) [inline lib/send-mail.ts]: Technically, this change does not fit the commit message, but I do not object strongly to it being in the same commit as the other `;`-related changes.

File: lib/patch-series.ts [hunk 4]
```diff
@@ -86,20 +86,20 @@ export class PatchSeries {
             match = tagMessage.match(/^[\s\S]*?\n\n([\s\S]*)/);
             (match ? match[1] : tagMessage).split("\n").map((line) => {
                 // tslint:disable-next-line:max-line-length
-                match = line.match(/https:\/\/lore\.kernel\.org\/.*\/([^\/]+)/);
```
Comment (dscho) [inline lib/patch-series.ts]: Oh right, this is my fault. I am _so_ used to escaping the slash in regular expressions that I escape it even within character ranges... Thanks for cleaning up my mess!

File: lib/mail-archive-helper.ts (hunk 1 above)
Comment (webstech) [reply lib/mail-archive-helper.ts]: Maybe next time.
```
const separator = "``````````\n";
return `${separator}${body}${body.endsWith("\n") ? "" : "\n"}${separator}`;
```

**Comment by webstech** [comment]
Thanks for the review.  Please go ahead with the merge.  I will work on my git skills for rolling back, updating and replaying history on my own time.  I had initially done all the changes at once and then broke them up into commits.  Maybe not the best strategy but there were a number of linting messages to get through.  The use of '+' caused numerous indent issues until I turned it off.

File: lib/mail-archive-helper.ts (hunk 1 above)
Comment (dscho) [reply lib/mail-archive-helper.ts]: :-)

**Comment by dscho** [comment]
> I had initially done all the changes at once and then broke them up into commits.

That's exactly how I perform similar tasks...

I'll just merge it, the shape of the commits is not _all_ that important ;-)

**Pull request merged**